File: ABC/ABC079/c.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string ABCD;
    cin >> ABCD;
    int num[4];
    for (int i=0; i<4; ++i) num[i] = stoi(ABCD.substr(i,1));
    char op[3];
    int ans;
    for (int i=0; i<(1<<3); ++i) {
        ans = num[0];
        for (int j=0; j<3; ++j) {
            if (i >> j & 1) {
                ans += num[j+1];
                op[j] = '+';
            } else {
                ans -= num[j+1];
                op[j] = '-';
            }
        }
        if (ans == 7) break;
    }

    cout << ABCD[0] << op[0] << ABCD[1] << op[1] << ABCD[2] << op[2] << ABCD[3] << "=7" << endl;
    return 0;
}

File: ABC/ABC079/c.py
ABCD = input()

for i in range(2 ** 3):
    ans = int(ABCD[0])
    pm = []
    for j in range(3):
        if (i >> j) & 1:
            ans += int(ABCD[j+1])
            pm.append("+")
        else:
            ans -= int(ABCD[j+1])
            pm.append("-")
    if ans == 7:
        print("{}{}{}{}{}{}{}=7".format(ABCD[0], pm[0], ABCD[1], pm[1], ABCD[2], pm[2], ABCD[3]))
        break